packages/overlay/src/OverlayTrigger.ts

```diff
@@ -77,9 +77,13 @@ export class OverlayTrigger extends LitElement {
     private targetContent?: HTMLElement;
 
     private handleClose(event?: CustomEvent<OverlayOpenCloseDetail>): void {
-        if (!event || this.open === event.detail.interaction) {
-            this.removeAttribute('open');
+        if (
+            event?.detail.interaction !== this.open &&
+            event?.detail.interaction !== this.type
```

Comment:
An overlay received either the content type or the overlay type as its `interaction` when opening, meaning that the interaction will be one of the current value of `open` or the value for `type` when a single overlay is being closed. 

Hover overlays _may_ close first and shouldn't close a second time. Particularly, this is important when a hover overlay is "closed" by opening a new overlay.